generator/testdata/rust/gclient/golden/iam/v1/src/client.rs

```diff
@@ -0,0 +1,89 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Code generated by sidekick. DO NOT EDIT.
+
+use crate::Result;
+use gax::error::Error;
+
+/// API Overview
+///
+/// Manages Identity and Access Management (IAM) policies.
+///
+/// Any implementation of an API that offers access control features
+/// implements the google.iam.v1.IAMPolicy interface.
+///
+/// ## Data model
+///
+/// Access control is applied when a principal (user or service account), takes
+/// some action on a resource exposed by a service. Resources, identified by
+/// URI-like names, are the unit of access control specification. Service
+/// implementations can choose the granularity of access control and the
+/// supported permissions for their resources.
+/// For example one database service may allow access control to be
+/// specified only at the Table level, whereas another might allow access control
+/// to also be specified at the Column level.
+///
+/// ## Policy Structure
+///
+/// See google.iam.v1.Policy
+///
+/// This is intentionally not a CRUD style API because access control policies
+/// are created and deleted implicitly with the resources to which they are
+/// attached.
+///
+/// # Notes
+///
+/// Application developers may use this trait to mock the  clients.
+///
+/// Services gain new RPCs routinely. Consequently, this trait gains new methods
+/// too. To avoid breaking applications the trait provides a default
+/// implementation for each method. These implementations return an error.
+#[async_trait::async_trait]
```

Comment:
Is this still needed for our use case? See: https://blog.rust-lang.org/2023/12/21/async-fn-rpit-in-traits.html#should-i-still-use-the-async_trait-macro

Should we be using the trait_variant?

Reply:
If we do need to use this, I wonder if this should be pub(crate) by default for our own needs and expose it with a feature for clients?

Reply:
> Is this still needed for our use case? See: https://blog.rust-lang.org/2023/12/21/async-fn-rpit-in-traits.html#should-i-still-use-the-async_trait-macro
> 
> Should we be using the trait_variant?

We are building against 1.70 (that was based on Tokio's MSRV), and `trait_variant` requires 1.75:

https://crates.io/crates/trait-variant

note that it is also 0.1.x

We can change the MSRV, or switch to using explicit futures.  The latter may be a more stable approach, and since it is generated code, the robots won't complain about extra typing.

Reply:
oops, I did not notice 1.70 for now. ack

Reply:
> https://crates.io/crates/trait-variant
>
> note that it is also 0.1.x

True, but at least the language team owns this which gives it more cred in my mind.